avm/res/app/managed-environment/main.bicep

```diff
@@ -321,8 +321,22 @@ type roleAssignmentType = {
 
 type certificateKeyVaultPropertiesType = {
   @description('Required. The resoource ID of the identity. This is the identity that will be used to access the key vault.')
-  identityResourceId: string
+  identity: string
```

Comment:
Hm ... as per the specs, this should actually say `resourceId` as it did before. In these cases, where the RP expects a different name, we'd usually map the value in the `resource` block. I.e.,:
```bicep
customDomainConfiguration: {
   certificatePassword: certificatePassword
   certificateValue: !empty(certificateValue) ? certificateValue : null
   dnsSuffix: dnsSuffix
   certificateKeyVaultProperties: {
      identity: certificateKeyVaultProperties.identityResourceId
      keyVaultUrl: certificateKeyVaultProperties.keyVaultUrl
   }
}
```